apps/src/lab2/levelEditors/levelData/EditMusicLevelData.tsx

```diff
@@ -252,6 +255,20 @@ const EditMusicLevelData: React.FunctionComponent<EditMusicLevelDataProps> = ({
             }}
             size="s"
           />
+          <Checkbox
+            checked={!!levelData.sortCdoSoundsByType}
+            name="sortCdoSoundsByType"
+            label="Sort Code.org sounds by type in Sound Picker"
+            onChange={event => {
+              const sortCdoSoundsByType = event.target.checked;
+              setLevelData({
+                ...levelData,
+                sortCdoSoundsByType,
+                ...(sortCdoSoundsByType ? {showSoundFilters: true} : {}),
```

Comment:
Optional: instead of handling the conditional logic in the setter, you can just assign the property here (and above on 234-237) and then use a `useEffect` to handle any dependent changes, i.e.

```
useEffect(() => {
  const {showSoundsPanelInSoundsMode, sortCdoSoundsByType, showSoundFilters} = levelData;
  if (sortCdoSoundsByType && !showSoundFilters) {
    setLevelData({...levelData, showSoundFilters: true});
  }
  if (!showSoundFilters && (sortCdoSoundsByType || showSoundsPanelInSoundsMode)) {
    setLevelData({...levelData, sortCdoSoundsByType: false, showSoundsPanelInSoundsMode: false});
  }
}, [levelData]);
```